### ISSUE TITLE
feat: Configuration options for aws_cloudtrail_events table

### ISSUE THREAD

**Issue by Vimao** [cloudquery/cloudquery#9252]
### Which problem is this feature request solving?

Currently when you launch cloudquery with collection of aws_cloudtrail_events table it seems to push all the events currently available in the account to the destination. This is often way too much, and it would be nice to have some control over the fetched data

### Describe the solution you'd like

Add configuration options:

Cloudtrail Event type whitelist/blacklist

Fetch all events not older than

This could help with alleviating the issue and allowing users more fine-grained control.

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by Vimao** [comment]
Hello @hermanschaaf 

Sure, i can explain in detail what i meant.

The aws_cloudtrail_events is an extremely interesting table, as it would allow us to correlate Cloudtrail Events with other data collected by Cloudquery and enrich the display by joining the information from multiple tables.

My use case was that i wanted to enrich the Spot Interruption Cloudtrail Events with metadata about the Ec2 Machines being interrupted. In the cloudtrail event there is only the instance_id, while i also need information like to which ASG this EC2 is attached and other metadata, so joining those 2 tables would be extremely helpful.

I would also like to collect events regarding degradation of hardware on AWS side and associate it with EC2 machines along with their metadata. 

Filtering for specific events would also allow for example filtering S3Digest events which are most of the time not super useful with a tool like Cloudquery, and it would also extremely decrease the first runtime for accounts with a lot of Cloudtrail events already on the account.

With filtering, the option to ingest only events not older than a specific amount of time would probably be redundant, but it could still be useful for older, larger accounts with a lot of cloudtrail events.

I actually think that most users will probably not need all the cloudtrail information collected by cloudquery, and would instead be interested in a dedicated subset of events, especially because of how much information cloudtrail generates.

**Comment by hermanschaaf** [comment]
Thanks for the details @Vimao, that helps a lot! Makes sense, let me see what I can do to enable this 👀

**Comment by nmeneil** [comment]
Hi,

We're in the same situation, with 50 accounts which have been running for years. I stopped the import at 180 million rows as it was nowhere near complete and I've now excluded that table. Being able to filter by time would help but having the ability to filter in/out events would be a massive win.

**Comment by michalz-rely** [comment]
similar story with the https://github.com/cloudquery/cloudquery/issues/9372

**Comment by vvatta** [comment]
We have 90 days history in cloudtrail but when I sync it takes days and never completes and fills up hundreds of GB of disk until it is filled up and we need only 1 day of logs. Feature would help a lot.

**Comment by hermanschaaf** [comment]
Hi all :wave: This feature was released yesterday as part of AWS plugin v17.3.0. It can be enabled in your CloudQuery config file like this:

```
kind: source
spec:
  name: aws
  # ... other standard options (refer to docs)
  spec:
    table_options:
      aws_cloudtrail_events:
        lookup_events:
          - lookup_attributes:
              - attribute_key: EventName
                attribute_value: RunInstances
          - lookup_attributes:
              - attribute_key: EventName
                attribute_value: StartInstances
```

All the filters that can be applied at the AWS API level can now also be passed through CloudQuery config. Please see [the docs](https://www.cloudquery.io/docs/plugins/sources/aws/configuration) for further details. We're marking this feature as experimental, because we're not sure whether this config format will be the final version. Your feedback will be very much appreciated!

Finally, kudos to @bbernays for the great work on this!
